### PR TITLE
mypy: Add create_model overload

### PR DESCRIPTION
**Pull request by uriyyo**
## Add create_model overload

Currently `mypy` will fail on this code:
```python
from pydantic import create_model

Foo = create_model("Foo")
```
```
error: Need type annotation for 'Foo'  [var-annotated]
```

With this PR `mypy` will correctly resolve return value type of `create_model`:
```python
from pydantic import create_model

Foo = create_model("Foo")
reveal_type(Foo)
```
```
note: Revealed type is 'Type[pydantic.main.BaseModel]'
```

And it will still work for a cases when `__base__` is specified:
```python
from pydantic import create_model, BaseModel


class Bar(BaseModel):
    a: int


Foo = create_model("Foo", __base__=Bar)
reveal_type(Foo)
```
```
note: Revealed type is 'Type[temp.Bar*]'
```

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

**Comment by uriyyo** [comment]
please review

File: pydantic/main.py [hunk 1]
```diff
@@ -938,6 +938,32 @@ def __repr_args__(self) -> 'ReprArgs':
...
+    __config__: Type[BaseConfig] = None,
+    __base__: None = None,
+    __module__: str = __name__,
+    __validators__: Dict[str, classmethod] = None,
+    **field_definitions: Any,
+) -> Type['BaseModel']:
+    pass
+
+
+@overload
```
Comment (sbv-csis) [inline pydantic/main.py:954]: Im not an expert by any stretch of the imagination, but isn't this overload the same as the final implementation - is that needed?

Comment (uriyyo) [reply pydantic/main.py:954]: It's required to have all overloads defined.

Please see more info here [PEP 484](https://www.python.org/dev/peps/pep-0484/#id45)

**Review by PrettyWood — APPROVED** [review]
Great! Thank you so much

**Pull request merged**